tests/unit/matrix/test_cubemeasure.py

```diff
@@ -1379,37 +1370,59 @@ def it_knows_its_baseline(self, request, counts_with_missings):
             np.array([[0.28571429], [0.71428571]])
         )
 
-    def it_knows_its_columns_margin(self, raw_weighted_counts):
+    def it_knows_its_column_bases_non_array_x(
+        self, raw_weighted_counts, _row_dimension_is_array_prop_
+    ):
+        _row_dimension_is_array_prop_.return_value = False
         weighted_cube_counts = _CatXCatWeightedCubeCounts(
             None, raw_weighted_counts, None, None
         )
 
-        assert weighted_cube_counts.columns_margin == pytest.approx(
-            np.array([9.9, 7.7, 5.5])
+        assert weighted_cube_counts.column_bases == pytest.approx(
+            np.array([[9.9, 7.7, 5.5], [9.9, 7.7, 5.5]])
         )
 
-    def it_knows_its_row_bases(self, request, raw_weighted_counts):
-        property_mock(
-            request,
-            _CatXCatWeightedCubeCounts,
-            "rows_margin",
-            return_value=np.array([8.8, 9.9]),
+    def it_knows_its_column_bases_array_x(
```

Comment:
can you explain me the meaning of the _x? and also why the name of the tests after this one have the _x_ in the names?

Reply:
oh i get it, but TBH they're very complicate to interpret. Can we find another way to name these tests?

Usually we use dimensionality but now that you are generalizing the array I can understand the reason of the names but for example when we have a 1D we can just use `it_knows_row_bases_for_array_types` or something similar. Sorry for the noise :D

Reply:
We could add `any`? Do you think `it_knows_its_column_bases_array_x_any` is clearer?

Reply:
i would add a for if it's right in english `it_knows_its_column_bases_for_array_x_any`